core/block.go

```diff
@@ -14,23 +18,145 @@ type Block struct {
 	// The StarkNet address of the sequencer who created this block
 	SequencerAddress *felt.Felt
 	// The time the sequencer created this block before executing transactions
-	Timestamp uint64
+	Timestamp *felt.Felt
 	// The number of transactions in a block
-	TransactionCount uint64
+	TransactionCount *felt.Felt
 	// A commitment to the transactions included in the block
 	TransactionCommitment *felt.Felt
 	// The number of events
-	EventCount uint64
+	EventCount *felt.Felt
 	// A commitment to the events produced in this block
 	EventCommitment *felt.Felt
 	// The version of the StarkNet protocol used when creating this block
-	ProtocolVersion uint64
+	ProtocolVersion *felt.Felt
 	// Extraneous data that might be useful for running transactions
 	ExtraData *felt.Felt
 }
 
-func (b *Block) Hash() *felt.Felt {
-	// Todo: implement pedersen hash as defined here
-	// https://docs.starknet.io/documentation/develop/Blocks/header/#block_hash
-	return nil
+type BlockHashMetaInfo struct {
+	First07Block             uint64     // First block that uses the post-0.7.0 block hash algorithm
+	UnverifiableRange        []uint64   // Range of blocks that are not verifiable
+	FallBackSequencerAddress *felt.Felt // The sequencer address to use for blocks that do not have one
+}
+
+func getBlockHashMetaInfo(chain string) (*BlockHashMetaInfo, error) {
+	switch chain {
+	case "SN_MAIN":
+		fallBackSequencerAddress, _ := new(felt.Felt).SetString("0x021f4b90b0377c82bf330b7b5295820769e72d79d8acd0effa0ebde6e9988bc5")
+		return &BlockHashMetaInfo{
+			First07Block:             883,
+			UnverifiableRange:        []uint64{},
```

Comment:
If there is no `UnverifiableRange` for a block, we don't need to initialize it. Then on line 85, we can check `if unverifiableRange != nil`.